NitroxServer/Communication/UdpServer.cs

```diff
@@ -36,11 +38,20 @@ public void Start()
 
             Thread thread = new Thread(Listen);
             thread.Start();
+
+            isStopped = false;
+        }
+
+        public void Stop()
+        {
+            isStopped = true;
+
+            server.Shutdown("Shutting down server...");
         }
         
         private void Listen()
         {
-            while (true)
+            while (!isStopped)
             {
                 // Pause reading thread and wait for messages.
                 server.MessageReceivedEvent.WaitOne();
```

Comment:
`WaitOne` blocked indefinitely right now so just add a 500 to it like so: `server.MessageReceivedEvent.WaitOne(500);`

Reply:
Adding a timeout here might unexpected results, the NetServer.Shutdown resets the WaitOne.

Reply:
I'm not sure if the thread can stop gracefully without a timeout. If it looks fine to you (like, no hanging processes) then we'll just keep an eye on it.